pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -544,10 +561,9 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
 // An error is returned if the PVC object's phase is not "Bound".
 func (dswp *desiredStateOfWorldPopulator) getPVCExtractPV(
 	namespace string, claimName string) (*v1.PersistentVolumeClaim, error) {
-	pvc, err :=
-		dswp.kubeClient.CoreV1().PersistentVolumeClaims(namespace).Get(context.TODO(), claimName, metav1.GetOptions{})
+	pvc, err := dswp.pvcLister.PersistentVolumeClaims(namespace).Get(claimName)
```

Comment:
I am not sure were there any specific reasons to connect with API server. Cc @jsafrane

Reply:
I think there are some challenges with using informers in kubelet:
- nodeauthorizer restricts which pvcs/pvs are allowed to be accessed based on which pods are scheduled to the node. I'm not sure if this works with informers.
- All kubelets watching all PVCs/PVs will have scalability issues.

Reply:
This. Watching and tracking updates to all the things has pretty bad implications for kubelet memory usage, but also with scalability of the API Servers.

Reply:
cc @liggitt

Reply:
@msau42 @endocrimes Do you have any suggestions on how to fix this issue?

Reply:
> nodeauthorizer restricts which pvcs/pvs are allowed to be accessed based on which pods are scheduled to the node. I'm not sure if this works with informers.

currently pv is restricted to `get` and pvc is restricted to `get` and `update status`

> All kubelets watching all PVCs/PVs will have scalability issues.

agreed

> I am not sure were there any specific (historic) reasons to connect with API server. Cc @jsafrane

@jsafrane mentioned the possibility of live gets being used to guarantee interlocks and avoid issues with kubelets acting based on stale informers